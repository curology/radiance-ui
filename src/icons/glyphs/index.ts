// @create-index

export { default as AcneBodySprayGlyph } from './acneBodySprayGlyph';
export { default as AcneBodyWashGlyph } from './acneBodyWashGlyph';
export { default as AcneFiveGlyph } from './acneFiveGlyph';
export { default as AcneGlyph } from './acneGlyph';
export { default as AcneOneGlyph } from './acneOneGlyph';
export { default as AcneSevenGlyph } from './acneSevenGlyph';
export { default as AcneThreeGlyph } from './acneThreeGlyph';
export { default as AhaBhaTonerSolutionGlyph } from './ahaBhaTonerSolutionGlyph';
export { default as AntiAgingAndAcneGlyph } from './antiAgingAndAcneGlyph';
export { default as AntiAgingGlyph } from './antiAgingGlyph';
export { default as AzelaicAcidGlyph } from './azelaicAcidGlyph';
export { default as BodySunscreenGlyph } from './bodySunscreenGlyph';
export { default as BreakGlyph } from './breakGlyph';
export { default as CheckBadgeGlyph } from './checkBadgeGlyph';
export { default as ChestGlyph } from './chestGlyph';
export { default as CleanserGlyph } from './cleanserGlyph';
export { default as ClindamycinGlyph } from './clindamycinGlyph';
export { default as ClockAlwaysGlyph } from './clockAlwaysGlyph';
export { default as ClockMostOfTheTimeGlyph } from './clockMostOfTheTimeGlyph';
export { default as ClockNeverGlyph } from './clockNeverGlyph';
export { default as ClockSometimesGlyph } from './clockSometimesGlyph';
export { default as CloggedPoresGlyph } from './cloggedPoresGlyph';
export { default as CombinationGlyph } from './combinationGlyph';
export { default as ConsultHistoryGlyph } from './consultHistoryGlyph';
export { default as CourseGlyph } from './courseGlyph';
export { default as CycleGlyph } from './cycleGlyph';
export { default as DarkSpotFormulaGlyph } from './darkSpotFormulaGlyph';
export { default as DarkSpotsGlyph } from './darkSpotsGlyph';
export { default as DesonideGlyph } from './desonideGlyph';
export { default as DexpanthenolGlyph } from './dexpanthenolGlyph';
export { default as DropperGlyph } from './dropperGlyph';
export { default as ExerciseGlyph } from './exerciseGlyph';
export { default as ExfoliatingMaskGlyph } from './exfoliatingMaskGlyph';
export { default as EyecreamGlyph } from './eyecreamGlyph';
export { default as FaceGlyph } from './faceGlyph';
export { default as FemaleGlyph } from './femaleGlyph';
export { default as FineLinesGlyph } from './fineLinesGlyph';
export { default as FirmnessGlyph } from './firmnessGlyph';
export { default as FlagCanadaGlyph } from './flagCanadaGlyph';
export { default as FlagUkGlyph } from './flagUKGlyph';
export { default as FlagUsaGlyph } from './flagUSAGlyph';
export { default as FormulaChangeGlyph } from './formulaChangeGlyph';
export { default as GlobeGlyph } from './globeGlyph';
export { default as GreenTeaGlyph } from './greenTeaGlyph';
export { default as HandWithCreamGlyph } from './handWithCreamGlyph';
export { default as HydratingMaskGlyph } from './hydratingMaskGlyph';
export { default as HydrocolloidBandageGlyph } from './hydrocolloidBandageGlyph';
export { default as HydrocortisoneGlyph } from './hydrocortisoneGlyph';
export { default as HydroquinoneCycleGlyph } from './hydroquinoneCycleGlyph';
export { default as HydroquinoneGlyph } from './hydroquinoneGlyph';
export { default as IntersexGlyph } from './intersexGlyph';
export { default as KojicAcidGlyph } from './kojicAcidGlyph';
export { default as LightbulbGlyph } from './lightbulbGlyph';
export { default as LipBalmGlyph } from './lipBalmGlyph';
export { default as LockGlyph } from './lockGlyph';
export { default as LunchGlyph } from './lunchGlyph';
export { default as MaleGlyph } from './maleGlyph';
export { default as MelasmaGlyph } from './melasmaGlyph';
export { default as MessageGlyph } from './messageGlyph';
export { default as MetronidazoleGlyph } from './metronidazoleGlyph';
export { default as MicellarWaterGlyph } from './micellarWaterGlyph';
export { default as MixIngredientsGlyph } from './mixIngredientsGlyph';
export { default as MoisturizerGlyph } from './moisturizerGlyph';
export { default as MoonGlyph } from './moonGlyph';
export { default as MortorGlyph } from './mortorGlyph';
export { default as NeckGlyph } from './neckGlyph';
export { default as NiacinamideGlyph } from './niacinamideGlyph';
export { default as NoMakeupGlyph } from './noMakeupGlyph';
export { default as NoneGlyph } from './noneGlyph';
export { default as OftenDryGlyph } from './oftenDryGlyph';
export { default as OftenOilyGlyph } from './oftenOilyGlyph';
export { default as OilCleanserGlyph } from './oilCleanserGlyph';
export { default as OtherGlyph } from './otherGlyph';
export { default as PackagingGlyph } from './packagingGlyph';
export { default as PalmTreeGlyph } from './palmTreeGlyph';
export { default as PillsGlyph } from './pillsGlyph';
export { default as RelaxGlyph } from './relaxGlyph';
export { default as ResveratrolGlyph } from './resveratrolGlyph';
export { default as RichMoisturizerGlyph } from './richMoisturizerGlyph';
export { default as RocketshipGlyph } from './rocketshipGlyph';
export { default as SelfieGlyph } from './selfieGlyph';
export { default as SleepGlyph } from './sleepGlyph';
export { default as StressGlyph } from './stressGlyph';
export { default as SunGlyph } from './sunGlyph';
export { default as SunMoonGlyph } from './sunMoonGlyph';
export { default as SunblockGlyph } from './sunblockGlyph';
export { default as SunscreenGlyph } from './sunscreenGlyph';
export { default as SuperbottleGlyph } from './superbottleGlyph';
export { default as SupportCenterGlyph } from './supportCenterGlyph';
export { default as TextureGlyph } from './textureGlyph';
export { default as TimeAlwaysGlyph } from './timeAlwaysGlyph';
export { default as TimeMostOfTheTimeGlyph } from './timeMostOfTheTimeGlyph';
export { default as TimeNeverGlyph } from './timeNeverGlyph';
export { default as TimeSometimesGlyph } from './timeSometimesGlyph';
export { default as TranexamicAcidGlyph } from './tranexamicAcidGlyph';
export { default as TretinoinGlyph } from './tretinoinGlyph';
export { default as VeryDryGlyph } from './veryDryGlyph';
export { default as VeryOilyGlyph } from './veryOilyGlyph';
export { default as VitaminCGlyph } from './vitaminCGlyph';
export { default as VitaminCSerumGlyph } from './vitaminCSerumGlyph';
export { default as WashFaceGlyph } from './washFaceGlyph';
export { default as WrinklesGlyph } from './wrinklesGlyph';
export { default as YesGlyph } from './yesGlyph';
export { default as ZincGlyph } from './zincGlyph';
export { default as ZincSoapBarGlyph } from './zincSoapBarGlyph';
