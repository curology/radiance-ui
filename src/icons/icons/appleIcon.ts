import { AppleIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(AppleIcon, AppleIcon, props);
