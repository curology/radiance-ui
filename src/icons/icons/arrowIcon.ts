import { ArrowIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(ArrowIcon, ArrowIcon, props);
