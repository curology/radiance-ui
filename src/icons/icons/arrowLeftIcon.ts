import { ArrowLeftIcon, ArrowLeftIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowLeftIcon, ArrowLeftIconSecondary, props);
