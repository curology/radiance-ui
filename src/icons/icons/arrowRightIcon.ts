import { ArrowRightIcon, ArrowRightIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowRightIcon, ArrowRightIconSecondary, props);
