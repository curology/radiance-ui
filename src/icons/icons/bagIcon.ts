import { BagIcon, BagIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(BagIcon, BagIconSecondary, props);
