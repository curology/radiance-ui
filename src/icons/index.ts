/*
 * Note: this file has to be manually updated if we add more types of icons.
 * We do not generate it via create-index
 */

export * from './emojis';
export * from './glyphs';
export * from './icons';
export * from './logos';
export * from './navIcons';
