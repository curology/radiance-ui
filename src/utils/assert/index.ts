import { isDefined } from '../isDefined';

function assert(value: boolean, message?: string): asserts value;
function assert<T>(
  value: T | null | undefined,
  message?: string,
): asserts value is T;
/**
 * @see https://gist.github.com/jamiebuilds/2a40f565cba317a80bed1eec127340f6
 */
function assert(value: unknown, message?: string) {
  if (value === false || !isDefined(value)) {
    throw new Error(message ?? 'Assertion failed');
  }
}

export { assert };
