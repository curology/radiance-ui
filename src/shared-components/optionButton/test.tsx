import React from 'react';
import { render, userEvent } from 'src/tests/testingLibraryHelpers';
import { assert } from 'src/utils/assert';

import { AcneGlyph } from '../../icons';

import { OptionButton } from './index';

describe('<OptionButton />', () => {
  describe('UI snapshots', () => {
    it('checkbox selected, without custom icon', () => {
      const { container } = render(
        <OptionButton
          selected
          onClick={() => undefined}
          text="checkbox text"
          optionType="checkbox"
          buttonType="primary"
        />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('radio unselected, with icon node prop', () => {
      const { container } = render(
        <OptionButton
          onClick={() => undefined}
          text="radio text"
          optionType="radio"
          icon={<AcneGlyph />}
          buttonType="secondary"
        />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('onClick callback', () => {
    it('is invoked when clicked', () => {
      const spy = jest.fn();
      const { container } = render(
        <OptionButton
          onClick={spy}
          text="checkbox text"
          optionType="checkbox"
        />,
      );

      assert(container.firstElementChild);

      userEvent.click(container.firstElementChild);
      expect(spy).toHaveBeenCalled();
    });
  });
});
