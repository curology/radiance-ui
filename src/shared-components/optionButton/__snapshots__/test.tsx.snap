// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionButton /> UI snapshots checkbox selected, without custom icon 1`] = `
.emotion-0 {
  border-radius: 4px;
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.05);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
  padding: 1.5rem;
  margin-bottom: 1rem;
  width: 100%;
  text-align: left;
}

.emotion-0:hover {
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.10);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border: 1px solid;
  border-color: #332E54;
  background: #FFFFFF;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 350ms;
  transition: all 350ms;
  border-radius: 4px;
  background: #332E54;
  border-color: #332E54;
}

.emotion-4 svg {
  opacity: 0;
  -webkit-transition: all 350ms;
  transition: all 350ms;
}

.emotion-4 svg {
  opacity: 1;
  color: #FFFFFF;
  fill: #FFFFFF;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin-left: 1rem;
}

.emotion-9 {
  color: #524D6E;
  line-height: 1.5;
}

<button
  class="emotion-0 emotion-1"
  role="checkbox"
  type="button"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.96 2l-9.7 9.42L.88 7.1 0 7.99l5.26 5.18L15.83 2.9l-.87-.9z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      <div
        class="emotion-9 emotion-10"
      >
        checkbox text
      </div>
    </div>
  </div>
</button>
`;

exports[`<OptionButton /> UI snapshots radio unselected, with icon node prop 1`] = `
.emotion-0 {
  border-radius: 4px;
  background-color: #FFFFFF;
  border: 1px solid #EDEDF0;
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.05);
  cursor: pointer;
  -webkit-transition: box-shadow 200ms;
  transition: box-shadow 200ms;
  padding: 1.5rem;
  margin-bottom: 1rem;
  width: 100%;
  text-align: left;
}

.emotion-0:hover {
  box-shadow: 0px 8px 24px rgba(52, 51, 82, 0.10);
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border: 1px solid;
  border-color: #332E54;
  background: #FFFFFF;
  width: 32px;
  height: 32px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 350ms;
  transition: all 350ms;
  border-radius: 50%;
  width: 48px;
  height: 48px;
}

.emotion-4 svg {
  opacity: 0;
  -webkit-transition: all 350ms;
  transition: all 350ms;
}

.emotion-4 svg {
  opacity: 1;
  color: #332E54;
  fill: #332E54;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  margin-left: 1rem;
}

.emotion-9 {
  color: #524D6E;
  line-height: 1.5;
}

<button
  class="emotion-0 emotion-1"
  role="radio"
  type="button"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="48"
        viewBox="0 0 48 48"
        width="48"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M21.993 10.023c0 1.123.896 2.033 2.012 2.033s2.02-.92 2.02-2.033A2.03 2.03 0 0024.006 8c-1.107 0-2.012.9-2.012 2.023zm1.203 0a.81.81 0 01.809-.813c.452 0 .818.358.818.813a.826.826 0 01-.818.823.818.818 0 01-.809-.823zm-1.203 27.944c0 1.123.896 2.033 2.012 2.033s2.02-.92 2.02-2.033a2.03 2.03 0 00-2.02-2.023c-1.107 0-2.012.9-2.012 2.023zm1.203 0a.81.81 0 01.809-.813c.452 0 .818.368.818.813a.826.826 0 01-.818.823.818.818 0 01-.809-.823zm14.793-12.04a2.027 2.027 0 01-2.021-2.033 2.016 2.016 0 114.032 0c0 1.113-.895 2.032-2.011 2.032zm0-2.846a.818.818 0 100 1.635.812.812 0 00.808-.822.803.803 0 00-.808-.813zM8 23.893c0 1.122.905 2.032 2.021 2.032 1.116 0 2.021-.92 2.021-2.032a2.025 2.025 0 00-2.02-2.023c-1.117 0-2.022.9-2.022 2.023zm1.203 0a.818.818 0 111.635.009.818.818 0 01-1.635-.01z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
    >
      <div
        class="emotion-9 emotion-10"
      >
        radio text
      </div>
    </div>
  </div>
</button>
`;
